### PR TITLE
Feature/e UI 3595 roles and access tab display judiciary section

### PR DESCRIPTION
**Pull request by kuda1992**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3595

### Change description ###
Add a `Caserole` mock
Add a `CaseRole` resolver
Add  `CaseRole` display table

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: api/workAllocation2/interfaces/caseRole.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export interface CaseRole {
+  name: string;
+  role: string;
+  location: string;
+  start: string;
+  end: string;
```
Comment (connorpgpmcelroy) [inline api/workAllocation2/interfaces/caseRole.ts:6]: Minor query is we already have a Location interface - is this not related to this? Also, would start and end not be Date types rather than string? Just wanted to ask in case this has been done quick and not considered

File: api/workAllocation2/roleService.mock.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import MockAdapter from 'axios-mock-adapter';
+import { CASEROLES } from './constants/roles.mock.data';
+
+export const init = (mock: MockAdapter) => {
+  const getCaseRolesUrl = /http:\/\/am-role-assignment-service-aat.service.core-compute-aat.internal\/cases\/[a-fA-F0-9]{16}/;
```
Comment (connorpgpmcelroy) [inline api/workAllocation2/roleService.mock.ts:5]: Minor query - This probably won't work on preview/demo environments. Does it need to?

File: src/cases/components/roles-and-access/roles-and-access.component.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  <exui-case-roles-table [roles]="roles" [roleType]="'judicial'" [caseDetails]="caseDetails"></exui-case-roles-table>
+
+  <p>
+    <a href="javascript:void(0)" [routerLink]="['/', 'role-access', 'allocate-role']" [queryParams]="{'userType': 'judicial'}"
+       class="govuk-link">Allocate a role</a>
+  </p>
+
+  <h2 class="govuk-heading-m">Legal Ops</h2>
+
+  <exui-case-roles-table [roles]="[]" [caseDetails]="caseDetails"></exui-case-roles-table>
```
Comment (connorpgpmcelroy) [inline src/cases/components/roles-and-access/roles-and-access.component.html:15]: Minor query - Shouldn't this component be under containers if it's using an internal component? That's not always the case but usually the conjecture from my understanding

File: src/cases/components/roles-and-access/roles-and-access.component.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Component, Input } from '@angular/core';
+import { CaseView } from '@hmcts/ccd-case-ui-toolkit';
+import { CaseRole } from '../../../../api/workAllocation2/interfaces/caseRole';
+
+@Component({
+  selector: 'exui-roles-and-access',
+  templateUrl: './roles-and-access.component.html',
+  styleUrls: ['./roles-and-access.component.scss']
```
Comment (connorpgpmcelroy) [inline src/cases/components/roles-and-access/roles-and-access.component.ts:8]: Minor query - Is this required? Will there be stylings involved?

**Review by connorpgpmcelroy — COMMENTED** [review]
Have no changes to request but have some minor queries

File: src/cases/components/roles-and-access/roles-and-access.component.html (hunk 3 above)
Comment (connorpgpmcelroy) [reply src/cases/components/roles-and-access/roles-and-access.component.html:15]: Ignore this comment - Saw that there was a container component for this

File: api/workAllocation2/interfaces/caseRole.ts (hunk 1 above)
Comment (kuda1992) [reply api/workAllocation2/interfaces/caseRole.ts:6]: @connorpgpmcelroy I am not sure at his moment of time. I just assumed that the location property will be a `string`. No  one has mentioned the type for `location`.

File: api/workAllocation2/roleService.mock.ts (hunk 2 above)
Comment (kuda1992) [reply api/workAllocation2/roleService.mock.ts:5]: This is just a mock will be replaced once we have an integration endpoint.

File: src/cases/components/roles-and-access/roles-and-access.component.ts (hunk 4 above)
Comment (kuda1992) [reply src/cases/components/roles-and-access/roles-and-access.component.ts:8]: I will remove 😏

File: src/cases/containers/roles-and-access-container/roles-and-access-container.component.ts [hunk 5]
```diff
@@ -1,15 +1,22 @@
...
 export class RolesAndAccessContainerComponent implements OnInit {
+  public roles: CaseRole[] = [];
+  public caseDetails: CaseView;
 
-  constructor() { }
+  constructor(private readonly route: ActivatedRoute) {
+  }
 
-  ngOnInit() {
+  public ngOnInit() {
```
Comment (rayliangatos) [inline src/cases/containers/roles-and-access-container/roles-and-access-container.component.ts:17]: missing void

File: src/role-access/components/case-roles-table/case-roles-table.component.html [hunk 6]
```diff
@@ -0,0 +1,39 @@
+<table class="govuk-table" *ngIf="roles &&  roles.length">
+  <thead class="govuk-table__head">
+  <tr class="govuk-table__row">
+    <th scope="col" class="govuk-table__header govuk-body govuk-!-font-size-19" style="width: 300px;">Name</th>
+    <th scope="col" class="govuk-table__header govuk-body govuk-!-font-size-19" style="width: auto;">Role</th>
+    <th scope="col" class="govuk-table__header govuk-body govuk-!-font-size-19" style="width: auto;">Location</th>
+    <th scope="col" class="govuk-table__header govuk-body govuk-!-font-size-19" style="width: auto;">Start</th>
+    <th scope="col" class="govuk-table__header govuk-body govuk-!-font-size-19" style="width: auto;">End</th>
+    <th scope="col" class="govuk-table__header govuk-body govuk-!-font-size-19" style="width: 100px;"></th>
```
Comment (rayliangatos) [inline src/role-access/components/case-roles-table/case-roles-table.component.html:9]: could all these style move to css

File: src/role-access/components/index.ts [hunk 7]
```diff
@@ -1,8 +1,10 @@
+import { RolesAndAccessComponent } from '../../cases/components/roles-and-access/roles-and-access.component';
```
Comment (rayliangatos) [inline src/role-access/components/index.ts:1]: unused import

File: src/role-access/components/index.ts [hunk 8]
```diff
@@ -1,8 +1,10 @@
...
 import { AnswersComponent } from './answers/answers.component';
+import { CaseRolesTableComponent } from './case-roles-table/case-roles-table.component';
```
Comment (rayliangatos) [inline src/role-access/components/index.ts:3]: unused import

**Review by rayliangatos — COMMENTED** [review]
small comments but all others look good

**Review by udaydenduluri33 — APPROVED** [review]
@kuda1992  Just verified the Component with the changes on 1225 Preview. LGTM Thanks.

**Review by rayliangatos — APPROVED** [review]

File: api/workAllocation2/interfaces/caseRole.ts (hunk 1 above)
Comment (connorpgpmcelroy) [reply api/workAllocation2/interfaces/caseRole.ts:6]: As it's the Node layer not sure it will be relevant anyway - can always be changed if needed later

File: api/workAllocation2/roleService.mock.ts (hunk 2 above)
Comment (connorpgpmcelroy) [reply api/workAllocation2/roleService.mock.ts:5]: ok sure

**Review by connorpgpmcelroy — APPROVED** [review]
Queries addressed so happy to approve
